scapy/contrib/automotive/obd/iid/iids.py

```diff
@@ -0,0 +1,134 @@
+# This file is part of Scapy
+# See http://www.secdev.org/projects/scapy for more information
+# Copyright (C) Andreas Korb <andreas.d.korb@gmail.com>
+# Copyright (C) Nils Weiss <nils@we155.de>
+# This program is published under a GPLv2 license
+
+from scapy.fields import FieldLenField, FieldListField, StrFixedLenField, \
+    ByteField, ShortField, FlagsField
+from scapy.packet import Packet
+
+
+# See https://en.wikipedia.org/wiki/OBD-II_PIDs#Service_09 for further information
+# IID = Information IDentification
+
+class OBD_IID00(Packet):
+    name = "IID_00_Service9SupportedInformationTypes"
+    fields_desc = [
+        FlagsField('supportedIIDs', b'', 32, [
+            'IID20',
+            'IID1F',
+            'IID1E',
+            'IID1D',
+            'IID1C',
+            'IID1B',
+            'IID1A',
+            'IID19',
+            'IID18',
+            'IID17',
+            'IID16',
+            'IID15',
+            'IID14',
+            'IID13',
+            'IID12',
+            'IID11',
+            'IID10',
+            'IID0F',
+            'IID0E',
+            'IID0D',
+            'IID0C',
+            'IID0B',
+            'IID0A',
+            'IID09',
+            'IID08',
+            'IID07',
+            'IID06',
+            'IID05',
+            'IID04',
+            'IID03',
+            'IID02',
+            'IID01'
+        ])
+    ]
+
+
+class _OBD_IID_MessageCount(Packet):
+    fields_desc = [
+        ByteField('messageCount', 0)
+    ]
+
+
+class OBD_IID01(_OBD_IID_MessageCount):
+    name = "IID_01_VinMessageCount"
+
+
+class OBD_IID03(_OBD_IID_MessageCount):
+    name = "IID_03_CalibrationIdMessageCount"
+
+
+class OBD_IID05(_OBD_IID_MessageCount):
+    name = "IID_05_CalibrationVerificationNumbersMessageCount"
+
+
+class OBD_IID07(_OBD_IID_MessageCount):
+    name = "IID_07_InUsePerformanceTrackingMessageCount"
+
+
+class OBD_IID09(_OBD_IID_MessageCount):
+    name = "IID_09_EcuNameMessageCount"
+
+
+class OBD_IID02(Packet):
+    name = "IID_02_VehicleIdentificationNumber"
+    fields_desc = [
+        ByteField('messageCount', 1),
+        StrFixedLenField('vehicleIdentificationNumber', b'', 17)
+    ]
+
+
+class OBD_IID04(Packet):
+    name = "IID_04_CalibrationId"
+    fields_desc = [
+        ByteField('messageCount', 1),
+        FieldListField('calibrationIdentifications', None,
+                       StrFixedLenField(b'', 0, 16),
+                       count_from=lambda pkt: len(pkt.original) / 16)
```

Comment:
It would be slightly cleaner to use a `//` with `count_from`. The API isn’t really thought for decimal values